src/feature/hs/hs_descriptor.c

```diff
@@ -1283,11 +1284,20 @@ cert_is_valid(tor_cert_t *cert, uint8_t type, const char *log_obj_type)
     log_warn(LD_REND, "Signing key is NOT included for %s.", log_obj_type);
     goto err;
   }
+
   /* The following will not only check if the signature matches but also the
    * expiration date and overall validity. */
   if (tor_cert_checksig(cert, &cert->signing_key, approx_time()) < 0) {
-    log_warn(LD_REND, "Invalid signature for %s: %s", log_obj_type,
-             tor_cert_describe_signature_status(cert));
+    if (cert->cert_expired) {
+      char expiration_str[ISO_TIME_LEN+1];
+      format_iso_time(expiration_str, cert->valid_until);
+      log_fn(LOG_PROTOCOL_WARN, LD_REND, "Invalid signature for %s: %s (%s)",
+             log_obj_type, tor_cert_describe_signature_status(cert),
+             expiration_str);
+    } else {
+      log_warn(LD_REND, "Invalid signature for %s: %s",
+               log_obj_type, tor_cert_describe_signature_status(cert));
```

Comment:
Wasn't the idea of the ticket to make this one proto warn? Because anyone can trigger that remotely or can they with your patch?

Reply:
It's true that people can trigger this remotely, but so is the case for hundreds of Tor's logs. AFAIK only the expiration case of this message is actually getting frequently triggered and that's why I only tried to hide that (and also add some more information for debugging).

I thought that by allowing the other messages get past, we might find a bug or something at some point.

Let me know and I can put everything at log proto warn.